modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/UpNextAdapter.kt

```diff
@@ -173,30 +173,43 @@ class UpNextAdapter(
                 }
 
                 shuffle.isVisible = FeatureFlag.isEnabled(Feature.UP_NEXT_SHUFFLE)
-                shuffle.setShuffleDisabledButton()
+                shuffle.updateShuffleButton()
 
                 shuffle.setOnClickListener {
-                    shuffle.setShuffleEnabledButton()
+                    settings.upNextShuffle.set(!settings.upNextShuffle.value, updateModifiedAt = false)
+                    shuffle.updateShuffleButton()
                 }
 
                 root.layoutParams.height = ViewGroup.LayoutParams.WRAP_CONTENT
             }
         }
 
-        private fun ImageButton.setShuffleDisabledButton() {
-            this.setImageDrawable(ContextCompat.getDrawable(context, IR.drawable.shuffle))
-            this.contentDescription = context.getString(LR.string.up_next_shuffle_button_content_description)
-            this.setImageTintList(ColorStateList.valueOf(ThemeColor.primaryIcon02(theme)))
+        private fun ImageButton.updateShuffleButton() {
+            if (!FeatureFlag.isEnabled(Feature.UP_NEXT_SHUFFLE)) return
 
-            TooltipCompat.setTooltipText(this, context.getString(LR.string.up_next_shuffle_button_content_description))
-        }
+            val isEnabled = settings.upNextShuffle.value
+
+            this.setImageDrawable(
+                ContextCompat.getDrawable(
+                    context,
+                    if (isEnabled) IR.drawable.shuffle_enabled else IR.drawable.shuffle,
+                ),
+            )
```

Comment:
```suggestion
            this.setImageResource(if (isEnabled) IR.drawable.shuffle_enabled else IR.drawable.shuffle)
```

Can you use `setImageResource` instead here?